hal/lib/athena/SerialHelper.cpp

```diff
@@ -183,75 +202,73 @@ void SerialHelper::QueryHubPaths(int32_t* status) {
-      }
+    std::shared_ptr<FILE> pipe(popen(val.c_str(), "r"), pclose);
+    // Just check the next item on a pipe failure
+    if (!pipe) continue;
+    while (!feof(pipe.get())) {
+      if (std::fgets(execBuffer, 128, pipe.get()) != 0) output += execBuffer;
+    }
+
+    if (!output.empty()) {
+      llvm::SmallVector<llvm::StringRef, 16> pathSplitVec;
+      // Split output by line, grab first line, and split it into
+      // individual directories
+      llvm::StringRef{output}.split('\n').first.split(pathSplitVec, '/', -1,
+                                                      false);
 
-      if (!output.empty()) {
-        llvm::SmallVector<llvm::StringRef, 16> pathSplitVec;
-        // Split output by line, grab first line, and split it into
-        // individual directories
-        llvm::StringRef{output}.split('\n').first.split(pathSplitVec, '/', -1,
-                                                        false);
-
-        // Find each individual item index
-
-        const char* usb1 = "usb1";
-        const char* tty = "tty";
-
-        int findusb = -1;
-        int findtty = -1;
-        int findregex = -1;
-        for (size_t i = 0; i < pathSplitVec.size(); i++) {
-          if (findusb == -1 && pathSplitVec[i].equals(usb1)) {
-            findusb = i;
-          }
-          if (findtty == -1 && pathSplitVec[i].equals(tty)) {
-            findtty = i;
-          }
-          if (findregex == -1 && pathSplitVec[i].equals(matchString)) {
-            findregex = i;
-          }
+      // Find each individual item index
+
+      const char* usb1 = "usb1";
+      const char* tty = "tty";
+
+      int findusb = -1;
+      int findtty = -1;
+      int findregex = -1;
+      for (size_t i = 0; i < pathSplitVec.size(); i++) {
+        if (findusb == -1 && pathSplitVec[i].equals(usb1)) {
+          findusb = i;
+        }
+        if (findtty == -1 && pathSplitVec[i].equals(tty)) {
+          findtty = i;
+        }
+        if (findregex == -1 && pathSplitVec[i].equals(matchString)) {
+          findregex = i;
         }
+      }
 
-        // Get the index for our device
-        int hubIndex = findtty;
-        if (findtty == -1) hubIndex = findregex;
+      // Get the index for our device
+      int hubIndex = findtty;
+      if (findtty == -1) hubIndex = findregex;
 
-        int devStart = findusb + 1;
+      int devStart = findusb + 1;
 
-        if (hubIndex < devStart) continue;
+      if (hubIndex < devStart) continue;
 
-        // Add our devices to our list
-        m_unsortedHubPath.push_back(
-            llvm::StringRef{pathSplitVec[hubIndex - 2]});
-        m_visaResource.push_back(llvm::StringRef{desc});
-        m_osResource.push_back(llvm::StringRef{matchString});
-      }
+      // Add our devices to our list
+      m_unsortedHubPath.push_back(llvm::StringRef{pathSplitVec[hubIndex - 2]});
+      m_visaResource.push_back(llvm::StringRef{desc});
+      m_osResource.push_back(llvm::StringRef{matchString});
```

Comment:
Do you need to call the constructor explicitly?  You might also be able to use emplace_back although I don't remember if SmallVector has it.